apiserver/facades/controller/sshtunneler/facade.go

```diff
@@ -45,3 +50,40 @@ func (f *Facade) RemoveSSHConnRequest(arg params.SSHConnRequestRemoveArg) (param
 	}
 	return params.ErrorResult{}, nil
 }
+
+// ControllerAddresses returns the specified machine's public addresses.
+func (f *Facade) ControllerAddresses(machine params.Entity) (params.StringsResult, error) {
+	mt, err := names.ParseMachineTag(machine.Tag)
+	if err != nil {
+		return params.StringsResult{Error: apiservererrors.ServerError(err)}, nil
+	}
+
+	m, err := f.backend.ControllerMachine(mt.Id())
+	if err != nil {
+		return params.StringsResult{Error: apiservererrors.ServerError(err)}, nil
+	}
+
+	var result params.StringsResult
+	result.Result = append(result.Result, m.Addresses().AllMatchingScope(network.ScopeMatchPublic).Values()...)
+	return result, nil
+}
+
+// MachineHostKeys returns the host keys for a specified machine.
+func (f *Facade) MachineHostKeys(machine params.SSHMachineHostKeysArg) (params.SSHPublicKeysResult, error) {
```

Comment:
nitpick: we don't need to return an error - it's never set to anything

Reply:
Good catch, will fix in a follow-up.